gdal/port/cpl_vsi.h

```diff
@@ -406,6 +406,55 @@ GByte CPL_DLL *VSIGetMemFileBuffer( const char *pszFilename,
+typedef char**         (*VSIFilesystemPluginReadDirCallback)       ( void *pUserData, const char *pszDirname, int nMaxFiles );
+typedef void*          (*VSIFilesystemPluginOpenCallback)          ( void *pUserData, const char *pszFilename, const char *pszAccess );
+typedef vsi_l_offset   (*VSIFilesystemPluginTellCallback)          ( void *pFile );
+typedef int            (*VSIFilesystemPluginSeekCallback)          ( void *pFile, vsi_l_offset nOffset, int nWhence );
+typedef size_t         (*VSIFilesystemPluginReadCallback)          ( void *pFile, void *pBuffer, size_t nSize, size_t nCount );
+typedef int            (*VSIFilesystemPluginReadMultiRangeCallback)( void *pFile, int nRanges, void ** ppData,
+                                                                     const vsi_l_offset* panOffsets, const size_t* panSizes );
+typedef VSIRangeStatus (*VSIFilesystemPluginGetRangeStatusCallback)( void *pFile, vsi_l_offset nOffset, vsi_l_offset nLength );
+typedef int            (*VSIFilesystemPluginEofCallback)           ( void *pFile );
+typedef size_t         (*VSIFilesystemPluginWriteCallback)         ( void *pFile, const void *pBuffer, size_t nSize,size_t nCount);
+typedef int            (*VSIFilesystemPluginFlushCallback)         ( void *pFile );
+typedef int            (*VSIFilesystemPluginTruncateCallback)      ( void *pFile, vsi_l_offset nNewSize );
+typedef int            (*VSIFilesystemPluginCloseCallback)         ( void *pFile );
+
+/* 
+    Callbacks are defined as a struct allocated by a call to VSIAllocFilesystemPluginCallbacksStruct
+    in order to try to maintain ABI stability when eventually adding a new member.
+    Any callbacks added to this struct SHOULD be added to the END of this struct
+*/
+typedef struct {
+    void                                        *pUserData;
+    VSIFilesystemPluginStatCallback             stat;
+    VSIFilesystemPluginUnlinkCallback           unlink;
+    VSIFilesystemPluginRenameCallback           rename;
+    VSIFilesystemPluginMkdirCallback            mkdir;
+    VSIFilesystemPluginRmdirCallback            rmdir;
+    VSIFilesystemPluginReadDirCallback          read_dir;
+    VSIFilesystemPluginOpenCallback             open;
+    VSIFilesystemPluginTellCallback             tell;
+    VSIFilesystemPluginSeekCallback             seek;
+    VSIFilesystemPluginReadCallback             read;
+    VSIFilesystemPluginReadMultiRangeCallback   read_multi_range;
+    VSIFilesystemPluginGetRangeStatusCallback   get_range_status;
+    VSIFilesystemPluginEofCallback              eof;
+    VSIFilesystemPluginWriteCallback            write;
+    VSIFilesystemPluginFlushCallback            flush;
+    VSIFilesystemPluginTruncateCallback         truncate;
+    VSIFilesystemPluginCloseCallback            close;
+} VSIFilesystemPluginCallbacksStruct;
+
+VSIFilesystemPluginCallbacksStruct* VSIAllocFilesystemPluginCallbacksStruct( void );
```

Comment:
this function and the below one should be CPL_DLL exported too.

Can you add some minimal Doxygen doc (in the .cpp) for those 3 functions ( with `@since GDAL 2.5`) ?